### ISSUE TITLE
Lighting problem

### ISSUE THREAD

**Issue by neko259** [freeminer/freeminer#186]
I've started a game with default mod and the "palm trees" or something with wide leaves at the top. I removed these leaves, placed them back and the texture was totally black.

**Comment by proller** [comment]
it was lighted after 5-20 seconds or stay black ?

**Comment by neko259** [comment]
@proller I've removed it and placed it again.
Now I've seen the same with many other nodes, including stairs, signs and particles after breaking a block. They seem to light after several seconds, but when I dig more and more into the cave, new areas are black for a time.

**Comment by proller** [comment]
too optimized [shit]lighting code...
now some light calculations skipped on client.
server have special light thread - and it can have calc lag depend on load

**Comment by HybridDog** [comment]
>server have special light thread - and it can have calc lag depend on load

that makes me know when the server accepts that l placed a node at some place, doesn't it?

**Comment by proller** [comment]
no, server assepts placed node almost immediatly, block marked as "need for light calc"
and then light thread calcuates light in this block when can, then block marked as "changed" and server sends it to you

**Comment by HybridDog** [comment]
ok, can l somehow make that the predicted nodes are e.g. coloured a bit red to find out when e.g. other players also see my house?

**Comment by HybridDog** [comment]
l just got a strange lighting bug:
![screenshot_20151112_190744](https://cloud.githubusercontent.com/assets/3192173/11126930/b4c9e7d0-8971-11e5-8452-2c65bff80175.png)
![screenshot_20151112_190748](https://cloud.githubusercontent.com/assets/3192173/11126943/c37f5eea-8971-11e5-92bb-c60471866ab4.png)

it's dark, then bright, then again dark …
